### PR TITLE
fix: read, read_unsafe, and add `_witness` for non-verified generator/hint result

### PR DESCRIPTION
**Pull request by kevjue**

### CODE REVIEW

**Review by puma314 — APPROVED** [review]

File: plonky2x/core/src/frontend/hash/keccak/mod.rs [hunk 1]
```diff
@@ -11,7 +11,8 @@ pub mod keccak256;
 
 impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
     /// WARNING: DO NOT USE IN PRODUCTION, this is unconstrained!
-    pub fn keccak256(&mut self, bytes: &[ByteVariable]) -> Bytes32Variable {
+    pub fn keccak256_witness(&mut self, bytes: &[ByteVariable]) -> Bytes32Variable {
+        // TODO: Need to constrain generator result
```
Comment (ratankaliani) [inline plonky2x/core/src/frontend/hash/keccak/mod.rs:15]: @puma314 Should we constrain this now?

File: plonky2x/core/src/frontend/eth/mpt/rlc.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
         b_offset: Variable,
         len: Variable,
     ) {
-        // TODO: implement
+        // TODO: Need to constrain generator result.
```
Comment (ratankaliani) [inline plonky2x/core/src/frontend/eth/mpt/rlc.rs:38]: @puma314 Should we constrain this now?

Comment (puma314) [reply plonky2x/core/src/frontend/eth/mpt/rlc.rs:38]: No that's fine, no one is going to use this file

File: plonky2x/core/src/frontend/hash/keccak/mod.rs (hunk 1 above)
Comment (puma314) [reply plonky2x/core/src/frontend/hash/keccak/mod.rs:15]: no, we don't have the implementation so fine to just leave as long as the name is changed to end with `_witness`

**Pull request merged**